Documentation/eliminating-pre-builts.md

```diff
@@ -0,0 +1,85 @@
+# Eliminating pre-builts in .NET repositories
+
+This is a detailed guide on how to eliminate pre-builts in a source-buildable repository.
+It is primarily intended for developers contributing to the `dotnet` organization.
+
+## Table of content
+  - [What is a Prebuilt](#what-is-a-prebuilt)
+  - [Elimitating pre-builts](#eliminating-pre-builts)
+  - [Allowed exceptions](#allowed-exceptions)
+  - [Contacts](#contacts)
+
+# What is a Prebuilt
+
+_Source-build_ is a process of building a given product on a single machine from source with no internet access.
+
+By definition, _pre-builts_ are dependencies that a repo has on binary files that are not built from source, where _build from source_ points to any package produced during the _current source-build_. Exceptions to this rule are dependencies that are picked up from the host distro such as `cmake`.
+In layman terms, this means that packages from `nuget.org`, Microsoft builds or other non-source-built binaries cannot be used for source-building a given repository.
+
+When onboarding a repository to source-build or adding a new dependency to a source-buildable one, the contributor runs the risk of introducing a new pre-built to the product. 
+To protect against this and catch any new pre-builts as soon as possible, Arcade source-build infrastructure provides _pre-built detection_ - MSBuild logic responsible for veryfing that no used dependency is a pre-built. In case one is discovered (for example, during a PR pipeline), the build will fail with an appropriate message somewhat similar to the following:
+
+```
+3 new packages used not in baseline! See report at ./artifacts/source-build/self/prebuilt-report/baseline-comparison.xml for more information. Package IDs are:
+  Microsoft.Bcl.AsyncInterfaces.8.0.0-alpha.1.22557.11  
+  Microsoft.Build.16.7.0
+  Microsoft.Build.Framework.14.3.0
+```
+
+Pre-built detection identifies the source of dependencies used to build the repository. 
+These dependencies include not only direct dependencies, but also build tooling as well as dangling dependencies - packages downloaded / used by tooling during the build process and not referenced by the project itself.
+Dependencies retrieved from external sources that are not explicitly excluded from pre-build detection will be flagged as pre-builts.
```

Comment:
I think we should standardize on "pre-built detection" vs "pre-build detection", it's back and forth in this doc a bit.

Reply:
That's on me, will scan the doc and fix any inconsistencies